src/AbstractNormalizedEventManager.php

```diff
@@ -32,6 +32,17 @@
     protected $eventCache;
 
     /**
+     * Detaches a listener from an event.
+     *
+     * @param string $event the event to attach too
+     * @param callable $callback a callable function
+     * @param int $priority The priority that was used to initially register the listener.
+     *
+     * @return bool True on success, false on failure
+     */
+    abstract public function detach($event, $callback, $priority = self::DEFAULT_PRIORITY);
```

Comment:
Why `abstract public` instead of `public function _zDetach()`? Descendants of this class are not guaranteed to implement an interface exposing `detach()`, and therefore should not be forced to do this. Only you know the use-case of the descendants.